internal/controller/apiextensions/composite/composition_functions.go

```diff
@@ -264,11 +299,56 @@ func (c *FunctionComposer) Compose(ctx context.Context, xr *composite.Unstructur
 			req.Input = in
 		}
 
-		// TODO(negz): Generate a content-addressable tag for this request.
-		// Perhaps using https://github.com/cerbos/protoc-gen-go-hashpb ?
-		rsp, err := c.pipeline.RunFunction(ctx, fn.FunctionRef.Name, req)
-		if err != nil {
-			return CompositionResult{}, errors.Wrapf(err, errFmtRunPipelineStep, fn.Step)
+		// used to store the requirements returned at the previous iteration
+		var requirements *v1beta1.Requirements
```

Comment:
```suggestion
		var previous *v1beta1.Requirements
```

Perhaps this would help document what it is? I can tell that it's Requirements from the type.

Reply:
we went with `d` and `fctx` for "storage" variables between steps, so I adopted the same pattern here. calling new the variables to store the temporary ones if needed.